etna/records/models.py

```diff
@@ -97,30 +352,23 @@ def custom_dimension_11(self) -> str:
     @property
     def custom_dimension_12(self) -> str:
         if self.level_code and self.level:
-            return "Level " + self.level_code + " - " + self.level
-        else:
-            return ""
+            return f"Level {self.level_code} - {self.level}"
+        return ""
 
     @property
     def custom_dimension_13(self) -> str:
-        if (
-            self.hierarchy_level3_reference_number
-            and self.hierarchy_level3_summary_title
-        ):
-            return (
-                self.hierarchy_level3_reference_number
-                + " - "
-                + self.hierarchy_level3_summary_title
-            )
-        else:
-            return ""
+        for ancestor in self.hierarchy:
+            if ancestor.level_code == 3:
+                if ancestor.has_reference_number():
+                    return f"{ancestor.reference_number} - {ancestor.title}"
+                return ancestor.title
+        return ""
 
     @property
     def custom_dimension_14(self) -> str:
-        if self.template_reference_number and self.template_summary_title:
-            return self.template_reference_number + " - " + self.template_summary_title
-        else:
-            return ""
+        if self.has_reference_number():
+            return f"{self.reference_number} - {self.title}"
+        return self.title
```

Comment:
As in the previous version of transforms we don't want to return a value if only one of them is present. Its both or blank.